python/cudf/cudf/core/multiindex.py

```diff
@@ -1733,3 +1733,33 @@ def _intersection(self, other, sort=None):
         if sort is None and len(other):
             return midx.sort_values()
         return midx
+
+    def _split_columns_by_levels(self, levels):
+        # This function assumes that for levels with duplicate names, they are
+        # specified by indices, not name by ``levels``. E.g. [None, None] can
+        # only be specified by 0, 1, not "None".
+
+        # Normalize named levels into indices
+        if levels is not None:
+            level_indices = set()
+            level_names = list(self.names)
+            for lv in levels:
+                if isinstance(lv, int):
+                    level_indices.add(lv)
+                else:
+                    level_indices.add(level_names.index(lv))
+        else:
+            level_indices = range(len(self._data))
+
+        # Split the columns
+        data_columns, index_columns = [], []
+        column_names, index_names = [], []
+        for i, (name, col) in enumerate(zip(self.names, self._data.columns)):
+            if i in level_indices:
+                name = f"level_{i}" if name is None else name
+                data_columns.append(col)
+                column_names.append(name)
+            else:
+                index_columns.append(col)
+                index_names.append(name)
+        return data_columns, index_columns, column_names, index_names
```

Comment:
The naming `column_names` is inconsistent since you have `data_columns` and `index_columns`. I would use `data_names` instead.